src/main/java/io/jawg/osmcontributor/ui/activities/PhotoActivity.java

```diff
@@ -217,9 +217,7 @@ protected void onCreate(Bundle savedInstanceState) {
         // Init parameters.
         latitude = getIntent().getDoubleExtra("latitude", 0);
         longitude = getIntent().getDoubleExtra("longitude", 0);
-        poiId = getIntent().getLongExtra("poiId", 0);
-        imageAdapter = new ImageAdapter(this, poiId);
-        gridPhotos.setAdapter(imageAdapter);
+        imageAdapters = new HashMap<Long, ImageAdapter>();
```

Comment:
I did this because by default, all pictures where added to the same adapter, making some pictures appearing even if not associated to a given POI, even when adapter was re-created when switching to another POI. I can try to make it work with a single adapter.